src/lwip/freertos/sys_arch.c

```diff
@@ -56,19 +56,80 @@ static StaticQueue_t gTCPIPMsgQueue;
 static uint8_t gTCPIPMsgQueueStorage[SYS_MESG_QUEUE_LENGTH * sizeof(void *)];
 #endif
 
+#if LWIP_FREERTOS_USE_STATIC_SEMAPHORES
+typedef struct
+{
+    StaticSemaphore_t semBuffer;
+    sys_sem_t * sem;
+} semPoolItem_t;
+
+static semPoolItem_t gSemPool[LWIP_FREERTOS_SEMAPHORE_POOL_ITEMS_NUM];
+#endif
+
 static inline u32_t TicksToMS(TickType_t ticks)
 {
     return (ticks * 1000) / configTICK_RATE_HZ;
 }
 
 void sys_init(void)
```

Comment:
This changes the impl for *all FreeRTOS* platforms, which is likely not what is intended.

Overall, I think this is a change that should be discussed in terms of why is `sys_arch.c` not provided by each platform as a porting layer (which is actually the usual method of porting lwip).

Reply:
The change is under `#ifdefs`, but if everyone who needs to instrument starts adding `#ifdef` to this common `sys_arch.c`, it will stop scaling quickly. Strongly recommend moving sys_arch.c under each platform, as is usually done for lwip porting.

Reply:
Moved this part to platform specific SDK.